lib/p2p/NodeList.ts

```diff
@@ -1,17 +1,38 @@
 import P2PRepository from './P2PRepository';
 import { NodeInstance, NodeFactory } from '../types/db';
-import { Address } from '../types/p2p';
+import { Address, ReputationEvent } from '../types/p2p';
 import addressUtils from '../utils/addressUtils';
+import { ReputationEventType } from '../types/enums';
+import { ms } from '../utils/utils';
+import { EventEmitter } from 'events';
+
+const ReputationEventWeight = {
+  [ReputationEventType.ManualBan]: -100000000,
+  [ReputationEventType.PacketTimeout]: -5,
+  [ReputationEventType.SwapTimeout]: -15,
+  [ReputationEventType.SwapSuccess]: 1,
+};
+
+// TODO: remove events after certain amount of time
+
+interface NodeList {
+  on(event: 'node.ban', listener: (nodePubKey: string, events: ReputationEvent[]) => void): this;
+  emit(event: 'node.ban', nodePubKey: string, events: ReputationEvent[]): boolean;
+}
 
 /** Represents a list of nodes for managing network peers activity */
-class NodeList {
+class NodeList extends EventEmitter {
+  private banThreshold = -50;
```

Comment:
This should be a static field

Reply:
Why `static`? Right now this variable is just used in the instance of `NodeList`. Setting it to `readonly` would make sense

Reply:
I think `static` makes more sense because it's an unchanging value that pertains to the class itself (e.g. all `NodeList`s should use this value) rather than a particular instance of `NodeList` (granted we're only using one). `readonly` from what I understand is for member fields that should be initialized once during the constructor and then not changed again.